### PR TITLE
Review of the Language Server Spec 

### LINKED ISSUE #707: Discrepancy in the `file/event` Notification Inputs Specification
### General Summary
We've observed a discrepancy within the file management API: the file/event notification is specified to have path and kind as inputs (see https://github.com/luna/enso/blob/master/doc/language-server/specification/enso-protocol.md#fileevent ), however the current implementation actually passes a single event input (that wraps path and kind) and this behavior seems to be covered by test (see https://github.com/luna/enso/blob/master/engine/language-server/src/test/scala/org/enso/languageserver/websocket/rpc/ReceivesTreeUpdatesHandlerTest.scala#L114 ). 

### Steps to Reproduce
Compare the linked files.

### Expected Result
Either specification should follow the implementation (and specify that the notification has a single `event` input) or the other way around.

Personally, I'd prefer the latter (having two inputs, instead of spurious type just to wrap them into `event`) but this doesn't make much difference.

### Actual Result
I see discrepancy.

### Enso Version
Latest master, i.e. 25494bbb4315d7d6d625424280dfdee0e49dd045

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #707 
related #708 

Changelog:

- fix: `file/event` notification parameters to match the spec
- fix: prefix all capabilities to match the spec
  + `canEdit` -> `text/canEdit`
  + `receivesTreeUpdates` -> `file/receivesTreeUpdates`
  + `canModify` -> `executionContext/canModify`
  + `receiveUpdates` -> `executionContext/receivesUpdates`
- remove: redundant `DirectoryTruncated` node and update the docs. Turned out we can indicate that directory have been truncated by returning `Directory` node in the `DirectoryTree` response.
- doc: fix `text/openFile`and `text/closeFile` wording
- doc: add `SHA3-224` description
- doc: rename Project Picker to Project Manager
- doc: rename capability receiveUpdates to receive**s**Updates to match the naming style (i.e. we have receivesTreeEvents)

### Important Notes

Changes in Spec:

- rename: `executionContext/receiveUpdates` to `executionContext/receivesUpdates`
- rename: `executionContext/create` result field `receivesEvents` to `receivesUpdates` matching the capability name
  ```
  {
    canModify: CapabilityRegistration;
    receivesUpdates: CapabilityRegistration;
  }
  ```
- rename: `executionContext/fork` result field `receivesEvents` to `receivesUpdates` matching the capability name
  ```
  {
    canModify: CapabilityRegistration;
    receivesUpdates: CapabilityRegistration;
  }
  ```
- remove: misleading `DirectoryTruncated` node and add specs about the truncated directories.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/style-guides/scala.md), [Java](https://github.com/luna/enso/blob/master/doc/style-guides/java.md), [Rust](https://github.com/luna/enso/blob/master/doc/style-guides/rust.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/style-guides/haskell.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

**Comment by iamrecursion** [comment]
Please update the doc with regards to the comment raised [here](https://github.com/luna/enso/issues/708#issuecomment-624073008).

File: doc/language-server/specification/enso-protocol.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ services components, as well as any open questions that may remain.
     - [`Position`](#position)
```
Comment (iamrecursion) [inline doc/language-server/specification/enso-protocol.md:66]: I want a clarification about how the file edit versioning scheme works somewhere in here.

Comment (4e6) [reply doc/language-server/specification/enso-protocol.md:66]: I added clarifications to the `FileEdit` section :+1:

File: engine/language-server/src/test/scala/org/enso/languageserver/requesthandler/monitoring/PingHandlerSpec.scala [hunk 2]
```diff
@@ -76,7 +76,7 @@ class PingHandlerSpec
     subsystem3.expectMsg(Ping)
     subsystem3.lastSender ! Pong
     //then
-    expectTerminated(actorUnderTest)
+    expectTerminated(actorUnderTest, 10.seconds)
```
Comment (4e6) [inline engine/language-server/src/test/scala/org/enso/languageserver/requesthandler/monitoring/PingHandlerSpec.scala:79]: This addresses the flaky test. PingHandler's timeout is set to 2 seconds, and the default AkkaTestkit timeout is 3 seconds. It leaves 1 second for the actor to terminate. For an overloaded build server, this is quite a small margin.

**Pull request merged**